### DISCUSSION TITLE
Are there plans to support more direct exporters?

### DISCUSSION THREAD

**Discussion by xuzhijvn** [open-telemetry/opentelemetry-go#3024] [Q&A] [answered]
### Problem Statement

Why do we need [opentelemetry-collector](https://github.com/open-telemetry/opentelemetry-collector)? Why not export directly? Although [opentelemetry-go](https://github.com/open-telemetry/opentelemetry-go) provides some exporters, such as jaeger, zipkin, stdout, etc., but there are too few, for example, there is no kafka exporter, I need it very much. I know the kafka exporter is available in [opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib), but as I said, I want to export directly instead of having to go through [opentelemetry-collector](https://github.com/open-telemetry/opentelemetry-collector). So, what exactly are its advantages?

**Comment by dmathieu** [comment, marked as answer]
First of all, nothing prevents you from writing your own exporter. It doesn't have to be within this repository or organization.

The collector is meant to be a receiver of all telemetry data, no matter the language of the service that sends it. So all implementations of OpenTelemetry will implement export into that format.
Then, if you wish to send the data to kafka, you need to configure the exporter in a single place, the exporter, not in every service.
If you wish to roll the credentials to the place where you're exporting data, you need to do it in a single service (the exporter), not your entire platform.

**Reply by xuzhijvn** [reply]
ok, thanks for answer
